### PR TITLE
Sync edit and delete actions with underlying permissions

### LINKED ISSUE #100: Add message when permissions prevent saving dandiset metadata
when you try to save a dandiset metadata and cannot due to permissions, we should print a message.

It might be better to not allow you to edit the dandiset metadata in the first place if you lack permissions. There could be desirable features on the editor page like downloading as yaml, but maybe you shouldn't get access to that anyway if you don't have permissions on the dandiset to edit.

### LINKED ISSUE #158: metadata edit icon should only be visible if the user has access
i logged in as a different user, and the edit icon is still visible.

image

this should only appear when:
1. a user is logged in
2. the user has access to edit the metadata

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Fixes #158.
Fixes #100. 

@satra This addresses both concerns outlined in that issue. While I think I covered all of the edge cases for deleting an items/folders, it would be good for you to double check this (obviously not by actually deleting things, but just ensuring that the delete button appears when you expect it to).

### CODE REVIEW

**Comment by satra** [comment]
@AlmightyYakob - indeed the button only appears when i am logged in and on a "my dataset" dataset. any chance the "actions" section could be moved up to the top of the panel in this PR? or do you prefer doing that in a separate PR?

**Comment by AlmightyYakob** [comment]
> @AlmightyYakob - indeed the button only appears when i am logged in and on a "my dataset" dataset. any chance the "actions" section could be moved up to the top of the panel in this PR? or do you prefer doing that in a separate PR?

Good to hear! As far as moving the actions up, that's definitely not an item for this PR, and might not be an easy fix, as the underlying component that we're using to display the data and actions only has a spot below the data for displaying actions.

**Comment by satra** [comment]
> the underlying component that we're using to display the data and actions only has a spot below the data for displaying actions.

i guess it's a girder component then. ok then for some future iteration.

File: web/src/views/DandisetLandingView/DandisetLandingView.vue [hunk 1]
```diff
@@ -201,6 +201,17 @@ export default {
   },
   computed: {
     loggedIn,
```
[the remark this answers is not in the document]
Comment (AlmightyYakob) [reply web/src/views/DandisetLandingView/DandisetLandingView.vue:203]: `loggedIn` is used in the template, so it needs to be a computed prop.

[the remark this answers is not in the document]
Comment (AlmightyYakob) [reply web/src/views/DandisetLandingView/DandisetLandingView.vue:203]: Nevermind you're right, I was on the wrong branch

File: web/src/components/MetaEditor.vue [hunk 2]
```diff
@@ -196,13 +214,17 @@ export default {
...
+      try {
+        const { status, data } = await girderRest.put(`folder/${this.id}/metadata`, { dandiset: this.meta });
+        if (status === 200) {
+          this.setCurrentDandiset(data);
+          this.closeEditor();
+        }
+      } catch ({ response: { status } }) {
+        if (status === 403) {
+          this.invalidPermissionSnackbar = true;
+        }
```
[the remark this answers is not in the document]
Comment (AlmightyYakob) [reply web/src/components/MetaEditor.vue:226]: It doesn't, I meant to add this, thanks for catching that.

File: web/src/views/DandisetLandingView/DandisetLandingView.vue [hunk 3]
```diff
@@ -201,6 +201,17 @@ export default {
...
+    editDisabledMessage() {
+      if (!this.loggedIn) {
+        return 'You must be logged in to edit.';
+      }
+
+      if (this.selected._accessLevel < 1) {
+        return 'You do not have permission to edit this dandiset.';
+      }
+
+      return null;
```
[the remark this answers is not in the document]
Comment (AlmightyYakob) [reply web/src/views/DandisetLandingView/DandisetLandingView.vue:213]: I'm not sure if there's an actual convention, but I tend to stray away from `undefined`, as to me it represents unintentional falseness. Here, we are explicitly setting this computed prop to `null` to signify that there's no error.

**Pull request merged**